shared/fetcherAnonymous.ts

```diff
@@ -0,0 +1,65 @@
+import axios from "axios";
+import Cookies from "js-cookie";
+
+import getConfig from "next/config";
+import { ITokenEnum } from "@shared/enum";
+
+const { publicRuntimeConfig } = getConfig();
+
+const baseURL = publicRuntimeConfig.BASE_URL;
+
+const api = axios.create({
+  baseURL,
+  timeout: 10000,
+  headers: { "content-type": "application/json"},
+});
+api.interceptors.request.use(
+  async (config) => {
+    const token = Cookies.get(ITokenEnum.anonymousToken);
+    if (token) {
+      config.headers.Authorization = `Bearer ${token}`;
+    }
+    return config;
+  },
+  (error) => Promise.reject(error)
+);
+
+api.interceptors.response.use(
+  (response) => response,
+  async (error) => {
+    if (
+      error?.response?.status === 401 &&
+      error?.response?.config?.url !== "/users/refresh-token"
+    ) {
+      const token = Cookies.get(ITokenEnum.refreshToken);
+      const refreshApi = axios.create({
+        baseURL,
+        timeout: 3000,
+        headers: {
+          "content-type": "application/json",
+          authorization: `Bearer ${token}`,
+        },
+      });
+      return refreshApi.get("/users/refresh-token").then(
+        async (refreshResponse) => {
+          const data = refreshResponse?.data?.data;
+          if (data?.token) {
+            Cookies.set(ITokenEnum.token, data?.token);
+            Cookies.set(ITokenEnum.refreshToken, data?.refresh_token);
+            Cookies.set(ITokenEnum.anonymousToken, data?.anonymousToken);
+            error.config.headers.Authorization = `Bearer ${data?.token}`;
+            return axios.request(error?.config);
+          }
+          return Promise.reject(error);
+        },
+        (refreshError) => {
+          console.log("[refreshError]: ", refreshError);
+          return Promise.reject(error);
+        }
+      );
+    }
+    return Promise.reject(error);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code handles the case when a request fails due to an expired token by trying to refresh the token and retrying the request. However, it doesn't handle the case where the refresh token request itself fails (other than logging the error). Consider adding a mechanism to redirect the user to the login page or show an error message when the refresh token request fails.
```diff
  (refreshError) => {
    console.log("[refreshError]: ", refreshError);
+   // Add logic here to handle refresh token failure
    return Promise.reject(error);
  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->